### PR TITLE
12.0 saas public website compatibility

### PR DESCRIPTION
**Pull request by trojikman**

### CODE REVIEW

File: saas_public/controllers/saas_public.py [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
-    @route('/saas_public/<model("saas.template"):template_id>/create-fast-build', type='http', auth='public')
+    @route('/saas_public/<int:template_id>/create-fast-build', type='http', auth='public')
     def create_fast_build(self, template_id, **kwargs):
         if not kwargs:
             kwargs = {}
-        if template_id and template_id.sudo().public_access:
-            template_operator_id = template_id.sudo().operator_ids.random_ready_operator()
+        if template_id:
+            template = request.env['saas.template'].browse(template_id)
+        if template.sudo().public_access:
```
Comment (yelizariev) [inline saas_public/controllers/saas_public.py]: You either don't check that ``template_id`` is specified or you check that template variable is set. I prefer the first option

File: saas_public/controllers/saas_public.py [hunk 2]
```diff
@@ -5,12 +5,14 @@
...
 class SaaSPublicController(Controller):
-    @route('/saas_public/<model("saas.template"):template_id>/create-fast-build', type='http', auth='public')
+    @route('/saas_public/<int:template_id>/create-fast-build', type='http', auth='public')
     def create_fast_build(self, template_id, **kwargs):
         if not kwargs:
             kwargs = {}
-        if template_id and template_id.sudo().public_access:
-            template_operator_id = template_id.sudo().operator_ids.random_ready_operator()
+        if template_id:
+            template = request.env['saas.template'].browse(template_id)
```
Comment (yelizariev) [inline saas_public/controllers/saas_public.py]: You can make sudo here once

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: saas_public/controllers/saas_public.py [hunk 3]
```diff
@@ -5,13 +5,14 @@
...
-    @route('/saas_public/<model("saas.template"):template_id>/create-fast-build', type='http', auth='public')
+    @route('/saas_public/<int:template_id>/create-fast-build', type='http', auth='public')
     def create_fast_build(self, template_id, **kwargs):
         if not kwargs:
             kwargs = {}
-        if template_id and template_id.sudo().public_access:
-            template_operator_id = template_id.sudo().operator_ids.random_ready_operator()
-            build = template_operator_id.sudo().create_db(kwargs, with_delay=False)
+        template = request.env['saas.template'].browse(template_id).sudo()
+        if template and template.public_access:
```
Comment (yelizariev) [inline saas_public/controllers/saas_public.py:13]: make ``else`` block and return something

Comment (trojikman) [reply saas_public/controllers/saas_public.py:13]: It is here https://github.com/it-projects-llc/saas-addons/blob/f77dbd99a2fca88065704ad089359a41d7bff755/saas_public/controllers/saas_public.py#L19

**Review by yelizariev — APPROVED** [review]

**Comment by yelizariev** [comment]
I approve to merge it now

**Pull request merged**